diff --git a/usr/share/applications/java-java-jdk.desktop b/usr/share/applications/java-java-jdk.desktop
index da271be..f0373ad 100644
--- a/usr/share/applications/java-java-jdk.desktop
+++ b/usr/share/applications/java-java-jdk.desktop
@@ -2,7 +2,7 @@
 Name=Oracle Java 17 Runtime
 Comment=Oracle Java 17 Runtime
 Keywords=java;runtime
-Exec=/usr/lib/jvm/java-17-jdk/bin/java -jar
+Exec=/usr/lib64/jvm/java-17-jdk/bin/java -jar
 Terminal=false
 Type=Application
 Icon=java-jdk
diff --git a/usr/share/applications/jconsole-java-jdk.desktop b/usr/share/applications/jconsole-java-jdk.desktop
index f73df6a..682c951 100644
--- a/usr/share/applications/jconsole-java-jdk.desktop
+++ b/usr/share/applications/jconsole-java-jdk.desktop
@@ -2,7 +2,7 @@
 Name=Oracle Java 17 Console
 Comment=Oracle Java 17 Monitoring and Management Console
 Keywords=java;console;monitoring
-Exec=/usr/lib/jvm/java-17-jdk/bin/jconsole
+Exec=/usr/lib64/jvm/java-17-jdk/bin/jconsole
 Terminal=false
 Type=Application
 Icon=java-jdk
diff --git a/usr/share/applications/jshell-java-jdk.desktop b/usr/share/applications/jshell-java-jdk.desktop
index 818ed8d..3fa22fd 100644
--- a/usr/share/applications/jshell-java-jdk.desktop
+++ b/usr/share/applications/jshell-java-jdk.desktop
@@ -2,7 +2,7 @@
 Name=Oracle Java 17 Shell
 Comment=Oracle Java 17 Shell
 Keywords=java;shell
-Exec=/usr/lib/jvm/java-17-jdk/bin/jshell
+Exec=/usr/lib64/jvm/java-17-jdk/bin/jshell
 Terminal=true
 Type=Application
 Icon=java-jdk
