--- a/include/toml11/fwd/region_fwd.hpp
+++ b/include/toml11/fwd/region_fwd.hpp
@@ -8,6 +8,9 @@
 
 #include <cassert>
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic warning "-Wuninitialized"
+
 namespace toml
 {
 namespace detail
@@ -102,3 +105,5 @@ class region
 } // namespace detail
 } // namespace toml
 #endif // TOML11_REGION_FWD_HPP
+
+#pragma GCC diagnostic pop
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -56,7 +56,7 @@ check_cxx_compiler_flag("-Wundef"
 check_cxx_compiler_flag("-Wshadow"              TOML11_COMPILER_SUPPORTS_WSHADOW)
 
 include(GNUInstallDirs)
-set(TOML11_INSTALL_CMAKE_DIR   ${CMAKE_INSTALL_LIBDIR}/cmake/toml11)
+set(TOML11_INSTALL_CMAKE_DIR   ${CMAKE_INSTALL_DATAROOTDIR}/cmake/toml11)
 set(TOML11_INSTALL_INCLUDE_DIR ${CMAKE_INSTALL_INCLUDEDIR})
 set(TOML11_CONFIG_DIR          ${CMAKE_CURRENT_BINARY_DIR}/cmake)
 set(TOML11_CONFIG              ${TOML11_CONFIG_DIR}/toml11Config.cmake)
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -153,6 +153,7 @@ if(TOML11_INSTALL)
     write_basic_package_version_file(${TOML11_CONFIG_VERSION}
         VERSION ${toml11_VERSION}
         COMPATIBILITY SameMajorVersion
+        ARCH_INDEPENDENT
         )
 
     configure_package_config_file(
