diff --git a/python/python3-tekore/python3-tekore.SlackBuild b/python/python3-tekore/python3-tekore.SlackBuild
index 92bdcb0..73a2ad9 100644
--- a/python/python3-tekore/python3-tekore.SlackBuild
+++ b/python/python3-tekore/python3-tekore.SlackBuild
@@ -81,6 +81,8 @@ find -L . \
  \( -perm 666 -o -perm 664 -o -perm 640 -o -perm 600 -o -perm 444 \
   -o -perm 440 -o -perm 400 \) -exec chmod 644 {} \;
 
+PYVER=$(python3 -c 'import sys; print("%d.%d" % sys.version_info[:2])')
+export PYTHONPATH=/opt/python$PYVER/site-packages/
 
 python3 -m build --wheel --no-isolation
 python3 -m installer --destdir="$PKG" dist/*.whl
diff --git a/python/python3-tekore/python3-tekore.info b/python/python3-tekore/python3-tekore.info
index 6c6ba68..594a3f5 100644
--- a/python/python3-tekore/python3-tekore.info
+++ b/python/python3-tekore/python3-tekore.info
@@ -5,6 +5,6 @@ DOWNLOAD="https://github.com/felix-hilden/tekore/archive/v5.4.0/tekore-5.4.0.tar
 MD5SUM="8a2b8b5b6bad6c64c055dfa39d2c634a"
 DOWNLOAD_x86_64=""
 MD5SUM_x86_64=""
-REQUIRES=""
+REQUIRES="python3-setuptools-opt"
 MAINTAINER="Pouria Rezaei"
 EMAIL="Pouria.rz@outlook.com"
