diff --git a/python/python3-tekore/python3-tekore.SlackBuild b/python/python3-tekore/python3-tekore.SlackBuild
index d3ba8cf..73a2ad9 100644
--- a/python/python3-tekore/python3-tekore.SlackBuild
+++ b/python/python3-tekore/python3-tekore.SlackBuild
@@ -81,6 +81,9 @@ find -L . \
  \( -perm 666 -o -perm 664 -o -perm 640 -o -perm 600 -o -perm 444 \
   -o -perm 440 -o -perm 400 \) -exec chmod 644 {} \;
 
+PYVER=$(python3 -c 'import sys; print("%d.%d" % sys.version_info[:2])')
+export PYTHONPATH=/opt/python$PYVER/site-packages/
+
 python3 -m build --wheel --no-isolation
 python3 -m installer --destdir="$PKG" dist/*.whl
 
