diff --git a/usr/share/applications/discord.desktop b/usr/share/applications/discord.desktop
index 5589311..3034bbc 100644
--- a/usr/share/applications/discord.desktop
+++ b/usr/share/applications/discord.desktop
@@ -3,7 +3,7 @@ Name=Discord
 StartupWMClass=discord
 Comment=All-in-one voice and text chat for gamers that's free, secure, and works on both your desktop and phone.
 GenericName=Internet Messenger
-Exec=/usr/share/discord/Discord
+Exec=/usr/lib64/discord/Discord
 Icon=discord
 Type=Application
 Categories=Network;InstantMessaging;
