diff --git a/input.txt b/input.txt
index d3a9268..996727f 100644
--- a/input.txt
+++ b/input.txt
@@ -58,7 +58,7 @@ pagedown PageDown
 delete-playlist W Ctrl
 delete-playlist-force W Shift Ctrl
 rename-playlist R Ctrl
-rename-playlist F2
+cycle-theme F2
 new-playlist T Ctrl
 
 new-generator-playlist E Ctrl Shift
diff --git a/t_modules/t_main.py b/t_modules/t_main.py
index 8492a61..056d295 100644
--- a/t_modules/t_main.py
+++ b/t_modules/t_main.py
@@ -4559,7 +4559,7 @@ try:
     gme.gme_open_file.restype = ctypes.c_char_p
 
 except:
-    print("Cannont find libgme")
+    pass
 
 def use_id3(tags, nt):
     def natural_get(tag, track, frame, attr):
